### PR TITLE
[RLlib] R2D2 training iteration fn AND switch off `execution_plan` API by default.

### PR DESCRIPTION
**Pull request by sven1977**
Switch off `execution_plan` API by default.
* Only a few algos remaining to be translated.
* R2D2 also done in this PR (shares already done DQN execution plan anyways).

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by gjoliver** [comment]
exciting! are these unit test failures real or known issues?

File: rllib/agents/maml/maml.py [hunk 1]
```diff
@@ -78,6 +78,9 @@
     # to tune vf_loss_coeff.
     # Use config.model.vf_share_layers instead.
     "vf_share_layers": DEPRECATED_VALUE,
+
```
Comment (sven1977) [inline rllib/agents/maml/maml.py:81]: Not done yet.

File: rllib/agents/mbmpo/mbmpo.py [hunk 2]
```diff
@@ -113,6 +113,9 @@
     # to tune vf_loss_coeff.
     # Use config.model.vf_share_layers instead.
     "vf_share_layers": DEPRECATED_VALUE,
+
```
Comment (sven1977) [inline rllib/agents/mbmpo/mbmpo.py:116]: Not done yet.

File: rllib/tests/test_rllib_train_and_evaluate.py [hunk 3]
```diff
@@ -38,8 +38,7 @@ def evaluate_test(algo, env="CartPole-v0", test_episode_rollout=False):
             + ', "min_sample_timesteps_per_reporting": 5,'
             '"min_time_s_per_reporting": 0.1, '
             '"model": {"fcnet_hiddens": [10]}'
-            "}' --stop='{\"training_iteration\": 1}'"
-            + " --env={} --no-ray-ui".format(env)
+            "}' --stop='{\"training_iteration\": 1}'" + " --env={}".format(env)
```
Comment (sven1977) [inline rllib/tests/test_rllib_train_and_evaluate.py:41]: small deprecation warning fix

File: rllib/agents/dqn/r2d2.py [hunk 4]
```diff
@@ -75,12 +75,6 @@
 
         # Update the target network every `target_network_update_freq` steps.
         "target_network_update_freq": 2500,
-
-        # Experimental flag.
```
Comment (avnishn) [inline rllib/agents/dqn/r2d2.py:79]: I'm assuming this just inherits from the dqn training iteration fn

**Review by avnishn — APPROVED** [review]
LGTM assuming R2D2 inherits from the dqn training itr fn

File: rllib/agents/dqn/r2d2.py (hunk 4 above)
Comment (sven1977) [reply rllib/agents/dqn/r2d2.py:79]: correct. R2D2 does not have its own `training_iteration` fn.

**Pull request merged**